hazelcast/src/main/java/com/hazelcast/util/ItemCounter.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Non Thread-Safe Counter of things. It allows to count items without worrying about nulls.
+ *
+ * @param <T>
+ */
+public final class ItemCounter<T> {
+    private final Map<T, Long> map = new HashMap<T, Long>();
```

Comment:
Maybe you can also use something like LongHolder instead of creating a new Long object in every `add` call.
`class LongHolder {
public long value
}`